sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -60,6 +60,8 @@
 
     /* To hold each receive work item to be processed.*/
     private final AtomicReference<SynchronousMessageSubscriber> synchronousMessageSubscriber = new AtomicReference<>();
+    /* To make sure only create one synchronousMessageSubscriber.*/
+    private final Object newSubscriberLock = new Object();
```

Comment:
If we're using a lock for writing, is there a reason to use AtomicReference still rather than this for reading/writing? I wonder because the compareAndSet(null, newVersion) was supposed to fulfill the same behaviour as the lock here.

Reply:
I was not changing it because I saw some other places also used this `SynchronousMessageSubscriber`, and that part of code are not locked. I only locked creation of this subscriber.
Anu have a better solution to not using a lock , so I'll try to use that way to keep consistently using Atomic.